pkg/comp-functions/functions/common/pdb.go

```diff
@@ -0,0 +1,64 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	pdbv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	ptr "k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	fnproto "github.com/crossplane/function-sdk-go/proto/v1beta1"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+)
+
+func AddPDBSettings(comp client.Object) func(ctx context.Context, svc *runtime.ServiceRuntime) *fnproto.Result {
+	return func(ctx context.Context, svc *runtime.ServiceRuntime) *fnproto.Result {
+
+		log := svc.Log
+
+		err := svc.GetObservedComposite(comp)
+		if err != nil {
+			return runtime.NewFatalResult(fmt.Errorf("can't get composite: %w", err))
+		}
+		pdb, ok := comp.(PDB)
+		if !ok {
```

Comment:
It would be great to catch this problem at build time and not at runtime. I thought about a solution here but nothing comes into my mind. @Kidswiss maybe you have an idea here?

Reply:
I tried to think of something during my first review, unfortunately I wasn't able to get a working idea.

I'll have another look now that all my other remarks have been implemented.

Reply:
Huh... okay it's actually rather simple:

Make the `PDB` interface also implement `client.Object`:

```go
type PDB interface {
	GetInstanceNamespace() string
	GetInstances() int
	GetPDBLabels() map[string]string
	client.Object
}
```

Accept `PDB` in the `AddPDBSettings` function:

```go
func AddPDBSettings(comp PDB) func(ctx context.Context, svc *runtime.ServiceRuntime) *fnproto.Result {
```

Remove the type assert and rename all `pdb` variables to `comp`. No more type assertion needed and the compiler will scream at you if you forget to implement the interface.

Reply:
perfect. Awesome. Thank for this :)

Reply:
the whole point I wanted to keep it in different interface was that not all services are ready for HA from the beginning, not all services even have a field `instances` in their claim. Now it's using InfoGetter{} as mentioned in the beginning and I completely got rid of PDB{} interface

Reply:
The idea of my comment wasn't to get rid of the `PDB` interface completely, just reduce the amount of duplication.

> not all services even have a field instances in their claim. 

In the long run we want all our service to have HA as well.